tests/unit/test_expression_tree/test_symbol.py

```diff
@@ -244,6 +249,13 @@ def test_evaluate_ignoring_errors(self):
         self.assertEqual(pybamm.t.evaluate_ignoring_errors(t=0), 0)
         self.assertIsNone(pybamm.Parameter("a").evaluate_ignoring_errors())
         self.assertIsNone(pybamm.StateVector(slice(0, 1)).evaluate_ignoring_errors())
+        self.assertIsNone(pybamm.StateVectorDot(slice(0, 1)).evaluate_ignoring_errors())
+
+        # with self.assertRaisesRegex(pybamm.ShapeError, "Cannot find shape"):
+        #     pybamm.inner(
+        #         pybamm.Matrix(np.zeros((2, 3))), pybamm.Matrix(np.ones((4, 5)))
+        #     ).evaluate_ignoring_errors()
```

Comment:
For some reason this test works on my computer but fails on GitHub actions. Locally it catches the `pybamm.ShapeError` but on GitHub it triggers the previous `ValueError` (e.g. see https://github.com/pybamm-team/PyBaMM/runs/7972858999?check_suite_focus=true). Any ideas?

Reply:
It could be something to do with whether `debug_mode` is True or False, it sometimes behaves differently if you run an individual test script vs the whole test suite

Reply:
Nvm. It did something very weird: when debugging it didn't access that line, but if I change it I get an error because `pybamm.ShapeError` is not raised.